src/morphology.cpp

```diff
@@ -13,10 +13,73 @@
 #include "readers/morphologyHDF5.h"
 #include "readers/morphologySWC.h"
 
+namespace {
+
+void buildChildren(const std::shared_ptr<morphio::Property::Properties> properties) {
+    {
+        const auto& sections = properties->get<morphio::Property::Section>();
+        auto& children = properties->_sectionLevel._children;
+
+        for (unsigned int i = 0; i < sections.size(); ++i) {
+            const int32_t parent = sections[i][1];
+            children[parent].push_back(i);
+        }
+    }
+
+    {
+        const auto& sections = properties->get<morphio::Property::MitoSection>();
+        auto& children = properties->_mitochondriaSectionLevel._children;
+
+        for (unsigned int i = 0; i < sections.size(); ++i) {
+            const int32_t parent = sections[i][1];
+            children[parent].push_back(i);
+        }
+    }
+}
+
+morphio::SomaType getSomaType(long unsigned int num_soma_points) {
+    switch (num_soma_points) {
+    case 0:
+        return morphio::SOMA_UNDEFINED;
+    case 1:
+        return morphio::SOMA_SINGLE_POINT;
+    case 2:
+        return morphio::SOMA_UNDEFINED;
+    default:
+        break;
+    }
+    return morphio::SOMA_SIMPLE_CONTOUR;
+}
+
+morphio::Property::Properties loadURI(const std::string& source, unsigned int options) {
+    const size_t pos = source.find_last_of('.');
+    if (pos == std::string::npos) {
+        throw(morphio::UnknownFileType("File has no extension"));
+    }
+
+    // Cross-platform check of file existance
+    std::ifstream file(source.c_str());
+    if (!file) {
+        throw(morphio::RawDataError("File: " + source + " does not exist."));
+    }
+
+    std::string extension = source.substr(pos);
+
+    if (extension == ".h5" || extension == ".H5") {
+        return morphio::readers::h5::load(source);
+    } else if (extension == ".asc" || extension == ".ASC") {
+        return morphio::readers::asc::load(source, options);
+    } else if (extension == ".swc" || extension == ".SWC") {
+        return morphio::readers::swc::load(source, options);
+    }
```

Comment:
`else if` is not needed here because of the returns.

Reply:
I'm going to leave it how it was.